metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -73,6 +74,7 @@ public WorkflowExecution call() {
       lock.unlock();
       return workflowExecution;
     }
+    lock.unlock();
```

Comment:
Would it be better to wrap the block above in a try ... finally so that the unlock statement is ALWAYS executed (even if an exception occurs)?
Then you can also remove the other unlock statement in line 74.

Reply:
makes sense. fixed